Marlin/src/HAL/HAL_STM32F1/HAL_spi_STM32F1.cpp

```diff
@@ -79,15 +79,25 @@ void spiBegin() {
  * @details
  */
 void spiInit(uint8_t spiRate) {
+  /**
+   * STM32F1 APB1 = 72MHz, APB2 = 36MHz, max SPI speed of this MCU if 18Mhz 
+   * STM32F1 has 3 SPI ports, SPI1 in APB1, SPI2/SPI3 in APB2
+   * so the minimum prescale of SPI1 is DIV4, SPI2/SPI3 is DIV2
+  */
+  #if SPI_DEVICE == 1
+    #define SPI_CLOCK_MAX SPI_CLOCK_DIV4
+  #else
+    #define SPI_CLOCK_MAX SPI_CLOCK_DIV2
+  #endif
   uint8_t  clock;
   switch (spiRate) {
-    case SPI_FULL_SPEED:    clock = SPI_CLOCK_DIV2 ; break;
+    case SPI_FULL_SPEED:    clock = SPI_CLOCK_MAX ;  break;
     case SPI_HALF_SPEED:    clock = SPI_CLOCK_DIV4 ; break;
     case SPI_QUARTER_SPEED: clock = SPI_CLOCK_DIV8 ; break;
     case SPI_EIGHTH_SPEED:  clock = SPI_CLOCK_DIV16; break;
     case SPI_SPEED_5:       clock = SPI_CLOCK_DIV32; break;
     case SPI_SPEED_6:       clock = SPI_CLOCK_DIV64; break;
-    default:                clock = SPI_CLOCK_DIV2; // Default from the SPI library
+    default:                clock = SPI_CLOCK_MAX;  // Default from the SPI library
```

Comment:
If its for the SD, you dont have to touch this default...

Reply:
How is it the rates are all shifted by one?

Reply:
good question :p may be for this SPI1 vs SPI2...

Reply:
To note: SPI_CLOCK_DIV2 = SPI_BAUD_PCLK_DIV_2   = SPI_CR1_BR_PCLK_DIV_2,   /**< f_PCLK/2 */

Reply:
![image](https://user-images.githubusercontent.com/145119/61580735-62b1e780-ab15-11e9-8fd7-d5298057c90a.png)

Reply:
so according to that, the commit might be correct, but.. eeprom can support very high frequencies... and our sdcard init code use the speed parameter... so i think the "default case" should be left to be able to try it...

Reply:
If the SD and EEPROM can operate at different frequencies then they should get separate settings. I expect that each of our HAL **wrappers for peripherals built on top of SPI** will include in their headers the SPI settings that they would prefer, and then we have to choose at compile-time what speeds are possible given how the buses are shared.

Reply:
capture of our eeprom description : 
![image](https://user-images.githubusercontent.com/145119/61584963-93b00d80-ab51-11e9-858e-bf51e75b07cf.png)

Reply:
> so i think the "default case" should be left to be able to try it...

The rest is worth merging, so I will do that. I did an audit of EEPROM code recently, and I also reviewed the SPI code at the same time, but not with this in mind. I'll take another look to see how consistent SPI is in actual usage.